src/test/java/com/compass/vinyl/VinylDbRecorderTest.java

```diff
@@ -0,0 +1,315 @@
+package com.compass.vinyl;
+
+import com.compass.vinyl.player.DatabaseRecordPlayer;
+import com.compass.vinyl.player.RecordPlayer;
+import com.compass.vinyl.serializer.JSONSerializer;
+import com.compass.vinyl.serializer.models.Animal;
+import com.compass.vinyl.serializer.models.Bird;
+import com.compass.vinyl.serializer.models.Lion;
+import com.compass.vinyl.serializer.models.Tiger;
+import org.junit.jupiter.api.*;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class VinylDbRecorderTest {
```

Comment:
Because the entire set of test cases is the same, can we come up with a single template with just the vinyl instance built differently - one with localfilesystem & another with database? That way any new test case we identify & add would be part of both the implementations.

Reply:
+1, makes sense. I will make the change

Reply:
Created abstract classes for `RecorderTest` and `RecordPlayerTest` and allowed classes to build instance with different set of `RecordPlayer` and  `RecordingConfig`